### PR TITLE
Give some more things logcontexts

### PR DESCRIPTION
**Pull request by richvdh**

### CODE REVIEW

File: synapse/handlers/user_directory.py [hunk 1]
```diff
@@ -108,11 +108,13 @@ def notify_new_event(self):
         if self._is_processing:
             return
 
-        self._is_processing = True
-        try:
-            yield self._unsafe_process()
-        finally:
-            self._is_processing = False
+        def process():
+            self._is_processing = True
```
[the remark this answers is not in the document]
Comment (richvdh) [reply synapse/handlers/user_directory.py]: where would you have it moved to?

[the remark this answers is not in the document]
Comment (richvdh) [reply synapse/handlers/user_directory.py]: oh, iswym. yes ok

[the remark this answers is not in the document]
Comment (richvdh) [reply synapse/handlers/user_directory.py]: done, though I'm not entirely sure I agree it's better. In particular if `run_as_background_process` fails to start the process for some reason (stack overflow, perhaps), then we'll get stuck with `_is_processing = True` forever.

[the remark this answers is not in the document]
Comment (erikjohnston) [reply synapse/handlers/user_directory.py]: Hmm, yes. We could move the `if self._is_processing` check inside the function, which would be basically the same as before except we're wrapping the entirety of the function in a `run_as_background_process`.

[the remark this answers is not in the document]
Comment (richvdh) [reply synapse/handlers/user_directory.py]: yeah, though `run_as_background_process` has enough stuff in it that I'd prefer to short-circuit it if possible.

**Pull request merged**